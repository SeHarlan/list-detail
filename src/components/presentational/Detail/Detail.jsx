import React from 'react';
import PropTypes from 'prop-types';

const Detail = ({ name, photoUrl, affiliation }) => (
  <section>
    <h2>{name}</h2>
    <figure>
      <img src={photoUrl} alt={name}/>
      <figcaption>{affiliation}</figcaption>
    </figure>
  </section>
);

Detail.propTypes = {
  name: PropTypes.string.isRequired,
  photoUrl: PropTypes.string.isRequired,
  affiliation: PropTypes.string.isRequired
};

export default Detail;
