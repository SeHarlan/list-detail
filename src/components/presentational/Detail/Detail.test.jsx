import React from 'react';
import { shallow } from 'enzyme';
import Detail from './Detail';

describe('detail component', () => {
  it('renders detail', () => {
    const wrapper = shallow(<Detail name="aang" photoUrl="url.com" affiliation="test aff"/>);
    expect(wrapper).toMatchSnapshot();
  });
});
