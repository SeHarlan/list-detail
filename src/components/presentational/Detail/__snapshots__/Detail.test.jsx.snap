// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`detail component renders detail 1`] = `
<section>
  <h2>
    aang
  </h2>
  <figure>
    <img
      alt="aang"
      src="url.com"
    />
    <figcaption>
      test aff
    </figcaption>
  </figure>
</section>
`;
